test/parallel/test-zlib-flush-drain.js

```diff
@@ -28,6 +28,9 @@ const ws = deflater._writableState;
 const beforeFlush = ws.needDrain;
 let afterFlush = ws.needDrain;
 
+deflater.on('data', () => {
+});
```

Comment:
Due to backpressure it won't be able to finish unless consumed.

Reply:
Why?

Reply:
flush will try to push additional data into the stream (or wait for pending writes to complete), if the buffer is already above the highwatermark it won't complete until something has been read.

Reply:
https://nodejs.org/api/zlib.html#zlib_zlib_flush_kind_callback

> Calling this only flushes data from the internal zlib state, and does not perform flushing of any kind on the streams level. Rather, it behaves like a normal call to .write(), i.e. **it will be queued up behind other pending writes and will only produce output when data is being read from the stream**.